dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/SystemActions.java

```diff
@@ -46,30 +51,52 @@ public SystemActions()
         super( "/system" );
     }
 
+    public ApiResponse getTask( String taskType, String taskId )
+    {
+        return get( "/tasks/" + taskType + "/" + taskId );
+    }
+
+    /**
+     * Waits until the task is completed and returns a response
+     *
+     * @param taskType
+     * @param taskId
+     * @return
+     */
     public ApiResponse waitUntilTaskCompleted( String taskType, String taskId )
     {
-        logger.info( "Waiting until task " + taskType + " with id " + taskId + "is completed" );
-        ApiResponse response = null;
-        boolean completed = false;
-        while ( !completed )
-        {
-            response = get( "/tasks/" + taskType + "/" + taskId );
-            response.validate().statusCode( 200 );
-            completed = response.extractList( "completed" ).contains( true );
-        }
+        Callable<Boolean> taskIsCompleted = () -> getTask( taskType, taskId )
+            .validateStatus( 200 )
+            .extractList( "completed" ).contains( true );
+
+        with()
+            .pollInterval( 1, TimeUnit.SECONDS )
```

Comment:
> why did you choose
> atMost(20, SECONDS) vs default 10s

I ran the tests many times and 10 seconds weren't enough to import for some of the runs. So just to avoid the flakes. 10 seconds is plenty for an import summary to be generated. 

> not ignoring exceptions vs ignoreExceptions

Ignoring exceptions is needed for the case when the response is null. Rest assured throws an exception then, which was also our original problem. I never saw task progress response being null, so didn't think there's a need to ignore exceptions there.

Reply:
Do you agree to explicitly set the `.pollDelay( 100, TimeUnit.MILLISECONDS )` so it will not be 1 second?

Reply:
Done